### PR TITLE
Add support for constraints

### LINKED ISSUE #18: Add support for foreign keys
Currently hammer ignores foreign keys so that e.g the following detects no diff.

from:
```sql
CREATE TABLE T2 (
  ID INT64,
);
```

to:
```sql
CREATE TABLE T2 (
  ID INT64,
  CONSTRAINT FK_T2_ID FOREIGN KEY (ID) REFERENCES T1 (ID)
);
```
expected:
```
ALTER TABLE T2 ADD CONSTRAINT FK_T2_ID FOREIGN KEY (ID) REFERENCES T1 (ID)
```

### PR DESCRIPTION
**Pull request by iffyio**
Fixes #18

### CODE REVIEW

**Comment by daichirata** [comment]
Thank you for making the PR. I apologize for not being able to review it right now, but I will review it as soon as possible.

**Comment by iffyio** [comment]
Hi @daichirata any updates on this if you have time to have a look at it?

File: internal/hammer/diff.go [hunk 1]
```diff
@@ -40,6 +40,17 @@ func NewDatabase(ddl DDL) (*Database, error) {
 			} else {
 				return nil, fmt.Errorf("cannot find ddl of table to apply index %s", stmt.Name)
 			}
+		case *spansql.AlterTable:
```
Comment (daichirata) [inline internal/hammer/diff.go:43]: I think adding a foreign key should be expressed in a create table, just like adding a column. So I don't think we need to support alter tables, what do you think?

**Review by daichirata — COMMENTED** [review]
I apologize for the late reply.

File: internal/hammer/diff_test.go [hunk 2]
```diff
@@ -389,6 +389,339 @@ CREATE INDEX idx_t3 ON t3(t3_1);
 				`CREATE INDEX idx_t3 ON t3(t3_1)`,
 			},
 		},
+		// Create table with constraint
```
Comment (daichirata) [inline internal/hammer/diff_test.go:392]: Thank you for adding tests! 🎉

File: internal/hammer/diff.go (hunk 1 above)
Comment (iffyio) [reply internal/hammer/diff.go:43]: Spanner's [DDL allows it](https://cloud.google.com/spanner/docs/data-definition-language#alter_table), but also retrieving a ddl from spanner relies on this feature unfortunately which makes supporting it a requirement. For example running `gcloud spanner databases ddl  describe db --instance=test` on a db with a table that has an FK, spanner returns two statements 1 `CREATE TABLE` and a following `ALTER TABLE` to add the FK constraint.

Comment (daichirata) [reply internal/hammer/diff.go:43]: I understand. Thank you very much for checking the detailed cases !

**Review by daichirata — APPROVED** [review]
Thank you for adding a foreign key support! Great work!

**Pull request merged**